cherrypy/test/test_http.py

```diff
@@ -20,13 +21,32 @@ def encode_multipart_formdata(files):
     """Return (content_type, body) ready for httplib.HTTP instance.
 
     files: a sequence of (name, filename, value) tuples for multipart uploads.
+    filename can be a string or a tuple ('filename string', 'encoding')
     """
     BOUNDARY = '________ThIs_Is_tHe_bouNdaRY_$'
     L = []
     for key, filename, value in files:
         L.append('--' + BOUNDARY)
-        L.append('Content-Disposition: form-data; name="%s"; filename="%s"' %
-                 (key, filename))
+
+        filename_encoding = None
+        if isinstance(filename, (list, tuple)):
+            filename, filename_encoding = filename
+        if filename_encoding:
+            if six.PY3:
```

Comment:
If at all possible, tests shouldn't switch Python versions. It should be possible to write universal code. Switching on Python versions adds complexity and potential pitfalls.